src/Validation/Constraint/IdentifiedBy.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolationException;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 4.0.0
+ */
+final class IdentifiedBy implements Constraint
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    public function __construct(string $id)
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function assert(Token $token)
+    {
+        $claimSet = $token->claims();
+
+        if (!$claimSet->has('jti') || $claimSet->get('jti') !== $this->id) {
+            throw new ConstraintViolationException(
```

Comment:
You always repeat a `has` + `get` check here - I suggest having something like a `get` method that always returns `null` on no value instead (simpler in this scenario)